roles/docker/templates/docker.service.j2

```diff
@@ -0,0 +1,21 @@
+[Unit]
+Description=Docker Application Container Engine
+Documentation=http://docs.docker.com
+After=docker.socket early-docker.target network.target
+Requires=docker.socket early-docker.target
+
+[Service]
+EnvironmentFile=-/run/flannel_docker_opts.env
+{% if http_proxy is defined and http_proxy != '' %}
+Environment="HTTP_PROXY={{ http_proxy }}"
+Environment="NO_PROXY={{ docker_proxy_exceptions }}"
+{% endif %}
+Environment="DOCKER_TMPDIR={{ docker_tmp_dir }}"
+
+MountFlags=slave
```

Comment:
This is just default for coreos https://github.com/docker/docker/blob/master/contrib/init/systemd/docker.service
Seems that it could cause some troubles though https://github.com/kubernetes/kubernetes/issues/4869
https://github.com/docker/machine/pull/3029